client/data/home/use-skip-current-view-mutation.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ * External dependencies
+ */
+import { useCallback } from 'react';
+import { useMutation, useQueryClient, UseMutationResult } from 'react-query';
+
+/**
+ * Internal dependencies
+ */
+import wp from 'calypso/lib/wp';
+import { useHomeLayoutQueryParams } from './use-home-layout-query-params';
+import { fetchHomeLayout, getCacheKey } from './use-home-layout-query';
+
+type ReminderDuration = '1d' | '1w' | null;
+
+interface Variables {
+	reminder: ReminderDuration;
+}
+
+interface Result extends UseMutationResult< void, unknown, Variables > {
+	skipCurrentView: ( reminder: ReminderDuration ) => void;
+}
+
+function useSkipCurrentViewMutation( siteId: number ): Result {
+	const queryClient = useQueryClient();
+	const { dev } = useHomeLayoutQueryParams();
+
+	const mutation = useMutation< void, unknown, Variables >(
+		async ( { reminder } ) => {
+			const data = await queryClient.fetchQuery(
+				getCacheKey( siteId ),
+				() => fetchHomeLayout( siteId, { dev } ),
```

Comment:
Here you should pass the entire `query` returned from `useHomeLayoutQueryParams()`, not just the `dev` part. That guarantees that both places where we execute the query (the hook and this async function) use exactly the same Query Function and that it doesn't matter which one is called first.

Reply:
Thanks for catching, I got confused because we don't want to send the `view` param to the skip API, so didn't think we'd want it here either.

> That guarantees that both places where we execute the query (the hook and this async function) use exactly the same Query Function and that it doesn't matter which one is called first.

👍